pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4208 @@
+              (classpath, class) objects that are representations of the
+              referencing association classes.
+
+              Each tuple represents one class and has these items:
+
+              * classpath (:class:`~pywbem.CIMClassName`): The class
+                path of the class, with its attributes set as follows:
+
+                * `classname`: Name of the class.
+                * `namespace`: Name of the CIM namespace containing the class.
+                * `host`: Host and optionally port of the WBEM server containing
+                  the CIM namespace.
+
+              * class (:class:`~pywbem.CIMClass`): The representation of the
+                class, with its `path` attribute set to the `classpath` tuple
+                item.
+
+        Raises:
+
+            CIMError: CIM_ERR_INVALID_NAMESPACE
+            CIMError: CIM_ERR_INVALID_PARAMETER
+        """  # noqa: E501
+        # pylint: enable=line-too-long
+
+        if isinstance(ObjectName, CIMInstanceName):
+            # This is an  instance reference
+            ref_paths = self._get_reference_instnames(namespace, ObjectName,
+                                                      ResultClass,
+                                                      Role)
+
+            instance_store = self.get_instance_store(namespace)
+            rtn_insts = [self._get_instance(namespace, path, instance_store,
+                                            None,
+                                            PropertyList, IncludeClassOrigin,
+                                            IncludeQualifiers)
+                         for path in ref_paths]
+
+            for inst in rtn_insts:
+                if inst.path.host is None:
+                    inst.path.host = self.host
+
+            return rtn_insts
+
+        rtn_classnames = self._get_reference_classnames(namespace, ObjectName,
+                                                        ResultClass, Role)
+        # returns list of tuples of (CIMClassname, CIMClass)
+        return self._return_assoc_class_tuples(rtn_classnames, namespace,
+                                               PropertyList, IncludeClassOrigin,
+                                               IncludeQualifiers)
+
+    ####################################################################
+    #
+    #   AssociatorNames server request
+    #
+    ####################################################################
+
+    def AssociatorNames(self, namespace, ObjectName, AssocClass=None,
+                        ResultClass=None, Role=None, ResultRole=None):
+        # pylint: disable=invalid-name
+        """
+        Implements a WBEM server responder for
+        :meth:`~pywbem.WBEMConnection.AssociatorNames`
+
+        Parameters:
+
+          namespace (:term:`string`):
+            Name of the namespace in which the CIM instance will be created
+              (case independent).
```

Comment:
Copy & paste error from CreateInstance. Should use enumeration verbiage.

Reply:
DONE.  It was massive copy error. They all now read the same and do not mention the actual operation.